tools/libboard/board_rc/board_rc.c

```
@@ -333,5 +346,125 @@ fpga_result print_board_info(fpga_token token)
+	char name[FPGA_STR_SIZE] = { 0 };
+
+	res = fpgaTokenGetObject(token, SYSFS_TCM_GLOB, &tcm_object, FPGA_OBJECT_GLOB);
+	if (res != FPGA_OK) {
+		OPAE_MSG("Failed to get token Object");
+		return res;
+	}
+
+	if (read_sysfs(token, SYSFS_TCM_BMC_FWVERS, name) == FPGA_OK) 
+		printf("BMC FW Version: %s", name);
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_BIP_VER, name) == FPGA_OK)
+		printf("BIP Version: %s", name);
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_FW_VER, name) == FPGA_OK)
+		printf("TCM FW Version: %s", name);
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_CRYPTO_VER, name) == FPGA_OK)
+		printf("Crypto block Version: %s", name);
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_SR_ROOT, name) == FPGA_OK)
+		printf("FIM root entry hash: %s", name);
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_BMC_ROOT, name) == FPGA_OK)
+		printf("BMC root entry hash: %s", name);
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_PR_ROOT, name) == FPGA_OK)
+		printf("PR root entry hash: %s", name);
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_BMC_FLASH_COUNT, name) == FPGA_OK)
+		printf("CBMC flash update counter: %s", name);
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_QSPI_COUNT, name) == FPGA_OK)
+		printf("User flash update counter: %s", name);
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_SR_CANCEL, name) == FPGA_OK)
+		printf("FIM CSK IDs canceled : %s", strlen(name) > 1 ? name : "None\n");
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_BMC_CANCEL, name) == FPGA_OK)
+		printf("BMC CSK IDs canceled: %s", strlen(name) > 1 ? name : "None\n");
+
+	memset_s(name, sizeof(name), 0);
+	if (read_sysfs(token, SYSFS_TCM_PR_CANCEL, name) == FPGA_OK)
+		printf("AFU CSK IDs canceled: %s", strlen(name) > 1 ? name : "None\n");
```

Silent errors when read_sysfs doesn't return FPGA_OK?

---

fixed